app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -225,6 +225,7 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
     fun showMegaTileError() {
         pocketVideosContainer.visibility = View.GONE
         pocketErrorContainer.visibility = View.VISIBLE
+
```

Comment:
(Copying @psymoon's comment into a review so we can have a discussion thread)

> Digging a little more deeper into this issue, it looks like Pocket should utilize the usage of ConnectivityManager. Only try pocket request if there's internet connection available. Try Again button also seems redundant since we are constantly trying to check for feed anyway (unless I'm mistaken). Calling requestFocus() within LiveData observer seems highly unstable and prone to race condition (especially with pocket request retries). This could be addressed further with FocusRepo implementation but for now I think we can force all focus to default if focus is lost. We could also look into different polling algorithms. What do you think, @Baron-Severin @dnarcese?

Reply:
> Pocket should utilize the usage of ConnectivityManager. Only try pocket request if there's internet connection available.

I don't have particularly strong feelings on this.  If the user has no internet connection, the requests instantly fail and otherwise have no impact on the user experience.  I would vote against the additional code complexity, but again, not a strong opinion.

> Try Again button also seems redundant since we are constantly trying to check for feed anyway (unless I'm mistaken).

It's definitely a narrow use case.  The backoff will eventually wait for a decent length of time between retries (the wait doubles between each attempt), it could be possible for a user to mess with their connection and then want to retry immediately.  The decision has UX ramifications as well, @aminalhazwani do you have any thoughts here?

> Calling requestFocus() within LiveData observer seems highly unstable and prone to race condition (especially with pocket request retries).

TBF, the requestFocus is within a click listener, which is a lot less concerning.

> We could also look into different polling algorithms.

For sure.  Also, the retry wait starts at 1 second, which is probably a little bit too eager.  There's an idea in A-C about breaking out a generic HTTP API client that handles this kind of stuff, so it would be good to refine our approach before writing that.  https://github.com/mozilla-mobile/android-components/issues/1508